ReSwift/CoreTypes/Middleware.swift

```diff
@@ -1,14 +1,99 @@
 //
 //  Middleware.swift
-//  ReSwift
+//  FPExamples
 //
-//  Created by Benji Encz on 12/24/15.
-//  Copyright © 2015 Benjamin Encz. All rights reserved.
+//  Created by Charlotte Tortorella on 25/11/16.
+//  Copyright © 2016 Charlotte Tortorella. All rights reserved.
 //
 
-import Foundation
+/**
+ Middleware is a structure that allows you to modify, filter out and dispatch more
+ actions, before the action being handled reaches the store.
+ */
+public struct Middleware {
+    public typealias DispatchFunction = (Action) -> Void
+    public typealias GetState = () -> StateType
 
-public typealias DispatchFunction = (Action) -> Any
-public typealias GetState = () -> StateType?
-public typealias Middleware =
-    (DispatchFunction?, @escaping GetState) -> (@escaping DispatchFunction) -> DispatchFunction
+    internal let transform: (GetState, DispatchFunction, Action) -> [Action]
+
+    /// Create a blank slate Middleware.
+    public init() {
+        self.transform = { [$2] }
```

Comment:
For readability I think it would be nice to have an explicit argument name for `action` here. 
E.g.

```swift
public init() {
    // The blank middleware just returns the untransformed action.
    self.transform = { _, _, action in 
           return action 
    }
}
```

While not needed for developers very familiar with ReSwift/FP I think it's a nice way to make it easier for beginners to understand the code/contribute.